prow/cmd/sub/main.go

```diff
@@ -149,12 +150,22 @@ func main() {
 	metrics.ExposeMetrics("sub", configAgent.Config().PushGateway, flagOptions.instrumentationOptions.MetricsPort)
 	pprof.Instrument(flagOptions.instrumentationOptions)
 
+	var cacheGetter subscriber.InRepoConfigCacheGetter
+	if flagOptions.cookiefilePath != "" {
+		cacheGetter = subscriber.InRepoConfigCacheGetter{
+			CookieFilePath: flagOptions.cookiefilePath,
+			CacheSize:      flagOptions.inRepoConfigCacheSize,
+			Agent:          configAgent,
+		}
+	}
+
 	s := &subscriber.Subscriber{
 		ConfigAgent:       configAgent,
 		InRepoConfigCache: cache,
```

Comment:
any chance this can be wrapped in the cache getter function below to avoid confusion?